scripts/linter_utils.py

```diff
@@ -68,12 +73,19 @@ def get_from_cache(key, factory):
             *. The result of factory(), or the last value to be associated to
             key.
         """
-        if key in cache:
-            return cache[key]
         with threadsafe_access(key):
             if key not in cache:
-                cache[key] = factory()
-        return cache[key]
+                # We need to use a container because variable assignment
```

Comment:
As mentioned in #6691, I don't really understand what's going on here. Can you add a link to a reference (and perhaps elaborate more on the explanation)?